### ISSUE TITLE
Lack of documentation

### ISSUE THREAD

**Issue by bergentroll** [Andrettin/Wyrmgus#162]
There is some docs, partly html, partly txt and a bit outdated. There may be cool to have short and clear README.md and also short, clean and generic build instruction with crucial options description.

**Comment by B4rabbas** [comment]
Yeah, I'm strugglin now to compile it, and the outated informations don't really help and made me perplex.

**Comment by bmorel** [comment]
Using Debian 10 stable "buster" and the git repo, I had no troubles compiling following instructions in the README.md file (not touched since 2019), but maybe it's because I'm too used to compile stuff.
Could you describe more the troubles you had to compile?

Those infos could also be valuable:

* Operating system (windows 7, windows8, windows 10, debian, centos, archlinux, freebsd, reactos, haiku, etc. I don't know what is really supported, so I put here a list of some OSes I know to exist :))
* compiler used (gcc, clang, Visual Studio, ...)

**Comment by joaocaju** [comment]
Not quite a documentation, but I managed to create a Steam Guide to create heroes on Wyrmsun - https://steamcommunity.com/sharedfiles/filedetails/?id=2728236434. 
It briefly mentions most-ish of the types of coding info for heroes.

**Comment by bmorel** [comment]
> Yeah, I'm strugglin now to compile it, and the outated informations don't really help and made me perplex.

Compilation instructions:

* If cloning the git repo, make sure to have done a recursive cloning. If not, the command `git submodule init; git submodule update --recursive` should fix it.
* create a build directory
* in the newly created build directory, run the command `cmake $path_to_wyrmsun_roo_directory`
* if cmake fails, you're missing dependencies, and cmake will have given you the name of the 1st missing one. Install it and re-run cmake until there are no more errors. ALternatively, you can read the build instructions to know which dependencies you need (dunno if it's outdated or not)
* then run `make`

It is possible to use other build engines than make, by giving cmake the information with the `-G` flag. For example, I often do `cmake .. -G Ninja`. It is possible to generate project files for visual studio, code::blocks, qtcreator and many others.

Those instructions could use a complete list of dependencies ofc. The current README.md file seems generated by some mean, so dunno what Andrettin would prefer.

**Comment by Andrettin** [comment]
@joaocaju Awesome guide! :)

**Comment by Andrettin** [comment]
@bmorel Thanks for writing the instructions! About the Wyrmsun readme file (if that's the one you meant), it was written by hand.

The dependencies are listed (for instance) here:
https://github.com/Andrettin/Wyrmgus/blob/master/doc/building_on_ubuntu_20.txt

**Comment by bmorel** [comment]
you're welcome. It's not that long to write after all. Have to say I have not checked this works on wyrmsun, but I see no reason why it would not, as this is my usual process when I clone a project that uses cmake (because i'm often too lazy to RTFM and get the deps list :D )
